CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -180,6 +180,12 @@ const char *_CFProcessPath(void) {
     }
     return __CFProcessPath;
 }
+
+#else
+
+Boolean _CFIsMainThread(void) {
+    return pthread_main_np() == 1;
+}
```

Comment:
I'm wondering if we should make that `#else` more platform specific. Other than that, this looks great to me.